### PR TITLE
Dockerize this project

### PR DESCRIPTION
**Pull request by ShiningRay**
added Dockerfile, docker-compose.yml and related files
upgraded Ruby to 3.0.3
upgraded related gems

### CODE REVIEW

**Comment by Keith-CY** [comment]
Should `tmp/` be committed?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+version: '3'
+
+services:
+  pg:
+    image: postgres:13-alpine
+    environment:
+      POSTGRES_USER: ckb
+      POSTGRES_PASSWORD: nervos
```
Comment (Keith-CY) [inline docker-compose.yml:8]: This should not be specified directly since forked repo will use them probably.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+ARG RUBY_VERSION=3.0.3
+FROM ruby:$RUBY_VERSION
+LABEL MAINTAINER ShiningRay
```
Comment (Keith-CY) [inline Dockerfile:3]: `MAINTAINER` should be `Nervos Network`

**Comment by ShiningRay** [comment]
> Should `tmp/` be committed?

Rails web application server will store pid files in `tmp/pids`, and put sockets file in `tmp/sockets`.
I checked the directories into git but ignore other files for deployment convenience

**Comment by Keith-CY** [comment]
> > Should `tmp/` be committed?
> 
> Rails web application server will store pid files in `tmp/pids`, and put sockets file in `tmp/sockets`. I checked the directories into git but ignore other files for deployment convenience

Gotcha

File: docker-compose.yml (hunk 1 above)
Comment (ShiningRay) [reply docker-compose.yml:8]: this docker-compose.yml file is used mainly for development.

File: app/models/ckb_sync/new_node_data_processor.rb [hunk 3]
```diff
@@ -117,9 +117,9 @@ def generate_tx_display_info(local_block)
     end
 
     def increase_records_count(local_block)
-      block_counter = TableRecordCount.find_by(table_name: "blocks")
+      block_counter = TableRecordCount.find_or_initialize_by(table_name: "blocks")
       block_counter.increment!(:count)
-      ckb_transaction_counter = TableRecordCount.find_by(table_name: "ckb_transactions")
+      ckb_transaction_counter = TableRecordCount.find_or_initialize_by(table_name: "ckb_transactions")
```
Comment (zmcNotafraid) [inline app/models/ckb_sync/new_node_data_processor.rb:122]: 这里不需要换成这个把？这样意思就变了，之前只是find, 现在find 不到会创建

File: db/schema.rb [hunk 4]
```diff
@@ -161,7 +161,7 @@
     t.integer "data_size"
     t.decimal "occupied_capacity", precision: 30
     t.decimal "block_timestamp", precision: 30
-    t.decimal "consumed_block_timestamp", precision: 30
+    t.decimal "consumed_block_timestamp", precision: 30, default: "0"
```
Comment (zmcNotafraid) [inline db/schema.rb:164]: 为什么会增加一个default

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

**Review by zmcNotafraid — APPROVED** [review]

File: app/models/ckb_sync/new_node_data_processor.rb (hunk 3 above)
Comment (ShiningRay) [reply app/models/ckb_sync/new_node_data_processor.rb:122]: 如果我从空数据库开始跑会报错，所以才改的

**Review by Keith-CY — APPROVED** [review]

**Pull request merged**